app/controllers/comments_controller.rb

```diff
@@ -1,6 +1,13 @@
 class CommentsController < ApplicationController
   def create
-    comment = post.comments.create(comment_params.merge(user_id: current_user.id))
-    respond_with post, comment
+    @post = Post.find_by_id(params[:post_id])
+    @post.comments.create(comment_params.merge(author_id: current_user.id))
+    redirect_back(fallback_location: root_path)
+  end
+
+  private
+
+  def comment_params
+    params.require(:data).permit(:text)
   end
```

Comment:
- [ ] The comments controller should have new actions similar to the one shown below. The `new` is used with the HTTP verb GET and creates an object instance.

```ruby

def new
    @comment = Comment.new
  end
```

Reply:
Done ☑️ .. Thanks for feedbacks

Reply:
well addressed this